core/src/com/unciv/logic/map/mapunit/UnitUpgradeManager.kt

```diff
@@ -12,15 +12,19 @@ class UnitUpgradeManager(val unit:MapUnit) {
 
     /** Returns FULL upgrade path, without checking what we can or cannot build currently.
      * Does not contain current baseunit, so will be empty if no upgrades. */
-    private fun getUpgradePath(): Iterable<BaseUnit> {
+    private fun getUpgradePath(upgradeUnit: BaseUnit?): Iterable<BaseUnit> {
```

Comment:
`null` returning an empty result is of no consequence I'm pretty sure. Both because this is now only called in scenarios where it wouldn't be null and because no unit to upgrade to. The only weirdo nonsense is I guess a lack of a check that `upgradeUnit` is in the upgrade path in the first place